tests/custom-scalars/src/test/kotlin/test/CustomScalarTest.kt

```diff
@@ -34,14 +45,23 @@ class CustomScalarTest {
         .query(GetAllQuery())
         .execute()
         .dataAssertNoErrors
-
+    assertEquals(1, data.id)
```

Comment:
Wait, how come that works? Isn't this comparing an `Int` with a `Long`?

Reply:
`assertEquals` uses `.equals` under the hood

Reply:
Yes but...

<img width="617" alt="Screenshot 2022-01-17 at 09 54 16" src="https://user-images.githubusercontent.com/3974977/149738157-a579524d-920a-4b65-9bc0-363e6896332b.png">

Reply:
🤔  Indeed!

Looks like a generics thing, but to be honest I'm surprised!

```kotlin
fun <T> isEqual(a: T, b: T): Boolean {
  println(a!!::class)
  println(b!!::class)
  return a == b
}

isEqual(1, 1L)
```
```
class kotlin.Long
class kotlin.Long
true
```

Reply:
TIL, Kotlin has implicit conversions for number literals: https://discuss.kotlinlang.org/t/at-which-situations-does-implicit-conversion-happen/16394/2

(And for the verbose version: https://kotlinlang.org/spec/built-in-types-and-their-semantics.html#integer-type-widening)

All in all, maybe use `assertEquals(1L, data.id)` to make it explicit that we're also checking the `Long` type here ?

Reply:
Very cool TIL too! And yes, totally agree :)